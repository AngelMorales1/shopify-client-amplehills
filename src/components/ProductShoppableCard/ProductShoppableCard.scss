@import 'settings';

.textInputAlcohol {
  font-size: 10px;
}

.margin {
  margin: 20px 0;
}

.round {
  position: relative;
  padding: 0;
  label {
    background-color: #fff;
    border: 2px solid #13466d;
    border-radius: 50%;
    cursor: pointer;
    height: 10px;
    left: 0;
    position: absolute;
    top: 0;
    width: 10px;
  }

  input[type='checkbox'] {
    visibility: hidden;
  }

  input[type='checkbox']:checked + label {
    background-color: #13466d;
    border-color: #13466d;
    &:after {
      opacity: 1;
    }
  }
}

.ProductShoppableCard {
  padding: spacing(4);

  @media ($medium-up) {
    padding: spacing(4) spacing(2) 0;
  }

  &__info {
    padding: spacing(6);
  }

  &__actions {
    height: $button-small-height;

    .ProductShoppableCard__quantity-selector {
      opacity: 0;
      transform: translate3d(0, spacing(-8), 0);
    }

    &--has-quantity {
      .ProductShoppableCard__quantity-selector {
        opacity: 1;
        transform: translate3d(0, 0, 0);
      }

      .ProductShoppableCard__button {
        opacity: 0;
        transform: translate3d(0, spacing(8), 0);
      }
    }
  }
}
