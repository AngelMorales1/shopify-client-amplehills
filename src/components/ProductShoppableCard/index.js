import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import productModel from 'models/productModel';

import { Button, QuantitySelector } from 'components/base';
import styles from './ProductShoppableCard.scss';
import get from 'utils/get';

const ProductShoppableCard = ({
  product,
  quantity,
  handleAddProduct,
  handleRemoveProduct
}) => {
  const handleQuantityChange = newQuantity => {
    if (newQuantity > quantity) handleAddProduct(product.handle);
    if (newQuantity < quantity) handleRemoveProduct(product.handle);
  };

  const actionClasses = cx(
    styles['ProductShoppableCard__actions'],
    'relative',
    {
      [styles['ProductShoppableCard__actions--has-quantity']]: quantity
    }
  );

  return (
    <div className={cx(styles['ProductShoppableCard'], 'flex col-12 sm-col-6')}>
      <div
        className="card flex flex-column w100"
        style={{
          background: `url(${get(product, 'gridImage', '')}) no-repeat center`,
          backgroundSize: 'cover'
        }}
      >
        <div className="aspect-5-4 w100" />
        <div
          className={cx(
            styles['ProductShoppableCard__info'],
            'flex flex-column justify-between bg-seafoam h100 relative'
          )}
        >
          <div>
            <span className="w100 bold mt2 mb1 block">
              {get(product, 'title', '')}
            </span>
            <p className="detail mb3">
              {get(product, 'flavorDescription', '')}
            </p>
          </div>
          {get(product, 'available', false) ? (
            <div className={actionClasses}>
              <QuantitySelector
                className={cx(
                  styles['ProductShoppableCard__quantity-selector'],
                  'absolute t0 l0 transition'
                )}
                quantity={quantity}
                color="seafoam"
                allowZero={true}
                onChange={newQuantity => handleQuantityChange(newQuantity)}
              />
              <Button
                className={cx(
                  styles['ProductShoppableCard__button'],
                  'small bg-seafoam absolute t0 l0'
                )}
                variant="primary-small"
                color="white-madison-blue-border"
                label="+ Add"
                onClick={() => handleAddProduct(product.handle)}
              />
            </div>
          ) : (
            <div className={actionClasses}>
              <Button
                className={cx(
                  styles['ProductShoppableCard__button'],
                  'small bg-seafoam absolute t0 l0'
                )}
                variant="primary-small"
                color="white-madison-blue-border"
                label="Sold Out"
                disabled={true}
              />
            </div>
          )}
        </div>
      </div>
    </div>
  );
};

ProductShoppableCard.propTypes = {
  product: productModel.propTypes,
  quantity: PropTypes.number,
  handleAddProduct: PropTypes.func,
  handleRemoveProduct: PropTypes.func
};

ProductShoppableCard.defaultProps = {
  product: productModel.defaultProps,
  quantity: 0,
  handleAddProduct: () => {},
  handleRemoveProduct: () => {}
};

export default ProductShoppableCard;
