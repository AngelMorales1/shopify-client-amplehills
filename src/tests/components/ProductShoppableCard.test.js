import React from 'react';
import { shallow } from 'enzyme';

import ProductShoppableCard from 'components/ProductShoppableCard';

it('renders without data', () => {
  const component = shallow(<ProductShoppableCard />);

  expect(component).toMatchSnapshot();
});

it('renders with data', () => {
  const component = shallow(
    <ProductShoppableCard
      product={{
        available: true,
        blocks: [],
        cartDetails: 'foo',
        description: 'foo',
        flavorDescription: 'foo',
        gridImage: 'foo',
        handle: 'foo',
        id: 'foo',
        pintImage: 'foo',
        price: 0,
        subItems: [],
        subItemsAvailable: true,
        title: 'foo',
        variants: [{}]
      }}
      quantity={0}
      handleAddProduct={() => console.log('foo')}
      handleRemoveProduct={() => console.log('foo')}
    />
  );

  expect(component).toMatchSnapshot();
});
