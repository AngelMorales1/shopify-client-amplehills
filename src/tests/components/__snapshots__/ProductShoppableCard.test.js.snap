// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with data 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProductShoppableCard
    handleAddProduct={[Function]}
    handleRemoveProduct={[Function]}
    product={
        Object {
            "available": true,
            "blocks": Array [],
            "cartDetails": "foo",
            "description": "foo",
            "flavorDescription": "foo",
            "gridImage": "foo",
            "handle": "foo",
            "id": "foo",
            "pintImage": "foo",
            "price": 0,
            "subItems": Array [],
            "subItemsAvailable": true,
            "title": "foo",
            "variants": Array [
              Object {},
            ],
          }
    }
    quantity={0}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="card flex flex-column w100"
        style={
                Object {
                        "background": "url(foo) no-repeat center",
                        "backgroundSize": "cover",
                      }
        }
>
        <div
                className="aspect-5-4 w100"
        />
        <div
                className="flex flex-column justify-between bg-seafoam h100 relative"
        >
                <div>
                        <span
                                className="w100 bold mt2 mb1 block"
                        >
                                foo
                        </span>
                        <p
                                className="detail mb3"
                        >
                                foo
                        </p>
                </div>
                <div
                        className="relative"
                >
                        <QuantitySelector
                                allowZero={true}
                                className="absolute t0 l0 transition"
                                color="seafoam"
                                onChange={[Function]}
                                quantity={0}
                                variant=""
                        />
                        <Button
                                className="small bg-seafoam absolute t0 l0"
                                color="white-madison-blue-border"
                                disabled={false}
                                fullWidth={false}
                                label="+ Add"
                                onClick={[Function]}
                                shadow={false}
                                to=""
                                type="button"
                                variant="primary-small"
                        />
                </div>
        </div>
</div>,
      "className": "flex col-12 sm-col-6",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="aspect-5-4 w100"
/>,
          <div
            className="flex flex-column justify-between bg-seafoam h100 relative"
>
            <div>
                        <span
                                    className="w100 bold mt2 mb1 block"
                        >
                                    foo
                        </span>
                        <p
                                    className="detail mb3"
                        >
                                    foo
                        </p>
            </div>
            <div
                        className="relative"
            >
                        <QuantitySelector
                                    allowZero={true}
                                    className="absolute t0 l0 transition"
                                    color="seafoam"
                                    onChange={[Function]}
                                    quantity={0}
                                    variant=""
                        />
                        <Button
                                    className="small bg-seafoam absolute t0 l0"
                                    color="white-madison-blue-border"
                                    disabled={false}
                                    fullWidth={false}
                                    label="+ Add"
                                    onClick={[Function]}
                                    shadow={false}
                                    to=""
                                    type="button"
                                    variant="primary-small"
                        />
            </div>
</div>,
        ],
        "className": "card flex flex-column w100",
        "style": Object {
          "background": "url(foo) no-repeat center",
          "backgroundSize": "cover",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "aspect-5-4 w100",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div>
                <span
                                className="w100 bold mt2 mb1 block"
                >
                                foo
                </span>
                <p
                                className="detail mb3"
                >
                                foo
                </p>
</div>,
              <div
                className="relative"
>
                <QuantitySelector
                                allowZero={true}
                                className="absolute t0 l0 transition"
                                color="seafoam"
                                onChange={[Function]}
                                quantity={0}
                                variant=""
                />
                <Button
                                className="small bg-seafoam absolute t0 l0"
                                color="white-madison-blue-border"
                                disabled={false}
                                fullWidth={false}
                                label="+ Add"
                                onClick={[Function]}
                                shadow={false}
                                to=""
                                type="button"
                                variant="primary-small"
                />
</div>,
            ],
            "className": "flex flex-column justify-between bg-seafoam h100 relative",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span
                    className="w100 bold mt2 mb1 block"
>
                    foo
</span>,
                  <p
                    className="detail mb3"
>
                    foo
</p>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "foo",
                    "className": "w100 bold mt2 mb1 block",
                  },
                  "ref": null,
                  "rendered": "foo",
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "foo",
                    "className": "detail mb3",
                  },
                  "ref": null,
                  "rendered": "foo",
                  "type": "p",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <QuantitySelector
                    allowZero={true}
                    className="absolute t0 l0 transition"
                    color="seafoam"
                    onChange={[Function]}
                    quantity={0}
                    variant=""
/>,
                  <Button
                    className="small bg-seafoam absolute t0 l0"
                    color="white-madison-blue-border"
                    disabled={false}
                    fullWidth={false}
                    label="+ Add"
                    onClick={[Function]}
                    shadow={false}
                    to=""
                    type="button"
                    variant="primary-small"
/>,
                ],
                "className": "relative",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "allowZero": true,
                    "className": "absolute t0 l0 transition",
                    "color": "seafoam",
                    "onChange": [Function],
                    "quantity": 0,
                    "variant": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": null,
                    "className": "small bg-seafoam absolute t0 l0",
                    "color": "white-madison-blue-border",
                    "disabled": false,
                    "fullWidth": false,
                    "label": "+ Add",
                    "onClick": [Function],
                    "shadow": false,
                    "to": "",
                    "type": "button",
                    "variant": "primary-small",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="card flex flex-column w100"
          style={
                    Object {
                              "background": "url(foo) no-repeat center",
                              "backgroundSize": "cover",
                            }
          }
>
          <div
                    className="aspect-5-4 w100"
          />
          <div
                    className="flex flex-column justify-between bg-seafoam h100 relative"
          >
                    <div>
                              <span
                                        className="w100 bold mt2 mb1 block"
                              >
                                        foo
                              </span>
                              <p
                                        className="detail mb3"
                              >
                                        foo
                              </p>
                    </div>
                    <div
                              className="relative"
                    >
                              <QuantitySelector
                                        allowZero={true}
                                        className="absolute t0 l0 transition"
                                        color="seafoam"
                                        onChange={[Function]}
                                        quantity={0}
                                        variant=""
                              />
                              <Button
                                        className="small bg-seafoam absolute t0 l0"
                                        color="white-madison-blue-border"
                                        disabled={false}
                                        fullWidth={false}
                                        label="+ Add"
                                        onClick={[Function]}
                                        shadow={false}
                                        to=""
                                        type="button"
                                        variant="primary-small"
                              />
                    </div>
          </div>
</div>,
        "className": "flex col-12 sm-col-6",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="aspect-5-4 w100"
/>,
            <div
              className="flex flex-column justify-between bg-seafoam h100 relative"
>
              <div>
                            <span
                                          className="w100 bold mt2 mb1 block"
                            >
                                          foo
                            </span>
                            <p
                                          className="detail mb3"
                            >
                                          foo
                            </p>
              </div>
              <div
                            className="relative"
              >
                            <QuantitySelector
                                          allowZero={true}
                                          className="absolute t0 l0 transition"
                                          color="seafoam"
                                          onChange={[Function]}
                                          quantity={0}
                                          variant=""
                            />
                            <Button
                                          className="small bg-seafoam absolute t0 l0"
                                          color="white-madison-blue-border"
                                          disabled={false}
                                          fullWidth={false}
                                          label="+ Add"
                                          onClick={[Function]}
                                          shadow={false}
                                          to=""
                                          type="button"
                                          variant="primary-small"
                            />
              </div>
</div>,
          ],
          "className": "card flex flex-column w100",
          "style": Object {
            "background": "url(foo) no-repeat center",
            "backgroundSize": "cover",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "aspect-5-4 w100",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div>
                  <span
                                    className="w100 bold mt2 mb1 block"
                  >
                                    foo
                  </span>
                  <p
                                    className="detail mb3"
                  >
                                    foo
                  </p>
</div>,
                <div
                  className="relative"
>
                  <QuantitySelector
                                    allowZero={true}
                                    className="absolute t0 l0 transition"
                                    color="seafoam"
                                    onChange={[Function]}
                                    quantity={0}
                                    variant=""
                  />
                  <Button
                                    className="small bg-seafoam absolute t0 l0"
                                    color="white-madison-blue-border"
                                    disabled={false}
                                    fullWidth={false}
                                    label="+ Add"
                                    onClick={[Function]}
                                    shadow={false}
                                    to=""
                                    type="button"
                                    variant="primary-small"
                  />
</div>,
              ],
              "className": "flex flex-column justify-between bg-seafoam h100 relative",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <span
                      className="w100 bold mt2 mb1 block"
>
                      foo
</span>,
                    <p
                      className="detail mb3"
>
                      foo
</p>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "foo",
                      "className": "w100 bold mt2 mb1 block",
                    },
                    "ref": null,
                    "rendered": "foo",
                    "type": "span",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "foo",
                      "className": "detail mb3",
                    },
                    "ref": null,
                    "rendered": "foo",
                    "type": "p",
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <QuantitySelector
                      allowZero={true}
                      className="absolute t0 l0 transition"
                      color="seafoam"
                      onChange={[Function]}
                      quantity={0}
                      variant=""
/>,
                    <Button
                      className="small bg-seafoam absolute t0 l0"
                      color="white-madison-blue-border"
                      disabled={false}
                      fullWidth={false}
                      label="+ Add"
                      onClick={[Function]}
                      shadow={false}
                      to=""
                      type="button"
                      variant="primary-small"
/>,
                  ],
                  "className": "relative",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "allowZero": true,
                      "className": "absolute t0 l0 transition",
                      "color": "seafoam",
                      "onChange": [Function],
                      "quantity": 0,
                      "variant": "",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "children": null,
                      "className": "small bg-seafoam absolute t0 l0",
                      "color": "white-madison-blue-border",
                      "disabled": false,
                      "fullWidth": false,
                      "label": "+ Add",
                      "onClick": [Function],
                      "shadow": false,
                      "to": "",
                      "type": "button",
                      "variant": "primary-small",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                ],
                "type": "div",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`renders without data 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProductShoppableCard
    handleAddProduct={[Function]}
    handleRemoveProduct={[Function]}
    product={undefined}
    quantity={0}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="card flex flex-column w100"
        style={
                Object {
                        "background": "url() no-repeat center",
                        "backgroundSize": "cover",
                      }
        }
>
        <div
                className="aspect-5-4 w100"
        />
        <div
                className="flex flex-column justify-between bg-seafoam h100 relative"
        >
                <div>
                        <span
                                className="w100 bold mt2 mb1 block"
                        />
                        <p
                                className="detail mb3"
                        />
                </div>
                <div
                        className="relative"
                >
                        <Button
                                className="small bg-seafoam absolute t0 l0"
                                color="white-madison-blue-border"
                                disabled={true}
                                fullWidth={false}
                                label="Sold Out"
                                onClick={[Function]}
                                shadow={false}
                                to=""
                                type="button"
                                variant="primary-small"
                        />
                </div>
        </div>
</div>,
      "className": "flex col-12 sm-col-6",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="aspect-5-4 w100"
/>,
          <div
            className="flex flex-column justify-between bg-seafoam h100 relative"
>
            <div>
                        <span
                                    className="w100 bold mt2 mb1 block"
                        />
                        <p
                                    className="detail mb3"
                        />
            </div>
            <div
                        className="relative"
            >
                        <Button
                                    className="small bg-seafoam absolute t0 l0"
                                    color="white-madison-blue-border"
                                    disabled={true}
                                    fullWidth={false}
                                    label="Sold Out"
                                    onClick={[Function]}
                                    shadow={false}
                                    to=""
                                    type="button"
                                    variant="primary-small"
                        />
            </div>
</div>,
        ],
        "className": "card flex flex-column w100",
        "style": Object {
          "background": "url() no-repeat center",
          "backgroundSize": "cover",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "aspect-5-4 w100",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div>
                <span
                                className="w100 bold mt2 mb1 block"
                />
                <p
                                className="detail mb3"
                />
</div>,
              <div
                className="relative"
>
                <Button
                                className="small bg-seafoam absolute t0 l0"
                                color="white-madison-blue-border"
                                disabled={true}
                                fullWidth={false}
                                label="Sold Out"
                                onClick={[Function]}
                                shadow={false}
                                to=""
                                type="button"
                                variant="primary-small"
                />
</div>,
            ],
            "className": "flex flex-column justify-between bg-seafoam h100 relative",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span
                    className="w100 bold mt2 mb1 block"
/>,
                  <p
                    className="detail mb3"
/>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "",
                    "className": "w100 bold mt2 mb1 block",
                  },
                  "ref": null,
                  "rendered": "",
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "",
                    "className": "detail mb3",
                  },
                  "ref": null,
                  "rendered": "",
                  "type": "p",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Button
                  className="small bg-seafoam absolute t0 l0"
                  color="white-madison-blue-border"
                  disabled={true}
                  fullWidth={false}
                  label="Sold Out"
                  onClick={[Function]}
                  shadow={false}
                  to=""
                  type="button"
                  variant="primary-small"
/>,
                "className": "relative",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": null,
                  "className": "small bg-seafoam absolute t0 l0",
                  "color": "white-madison-blue-border",
                  "disabled": true,
                  "fullWidth": false,
                  "label": "Sold Out",
                  "onClick": [Function],
                  "shadow": false,
                  "to": "",
                  "type": "button",
                  "variant": "primary-small",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="card flex flex-column w100"
          style={
                    Object {
                              "background": "url() no-repeat center",
                              "backgroundSize": "cover",
                            }
          }
>
          <div
                    className="aspect-5-4 w100"
          />
          <div
                    className="flex flex-column justify-between bg-seafoam h100 relative"
          >
                    <div>
                              <span
                                        className="w100 bold mt2 mb1 block"
                              />
                              <p
                                        className="detail mb3"
                              />
                    </div>
                    <div
                              className="relative"
                    >
                              <Button
                                        className="small bg-seafoam absolute t0 l0"
                                        color="white-madison-blue-border"
                                        disabled={true}
                                        fullWidth={false}
                                        label="Sold Out"
                                        onClick={[Function]}
                                        shadow={false}
                                        to=""
                                        type="button"
                                        variant="primary-small"
                              />
                    </div>
          </div>
</div>,
        "className": "flex col-12 sm-col-6",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="aspect-5-4 w100"
/>,
            <div
              className="flex flex-column justify-between bg-seafoam h100 relative"
>
              <div>
                            <span
                                          className="w100 bold mt2 mb1 block"
                            />
                            <p
                                          className="detail mb3"
                            />
              </div>
              <div
                            className="relative"
              >
                            <Button
                                          className="small bg-seafoam absolute t0 l0"
                                          color="white-madison-blue-border"
                                          disabled={true}
                                          fullWidth={false}
                                          label="Sold Out"
                                          onClick={[Function]}
                                          shadow={false}
                                          to=""
                                          type="button"
                                          variant="primary-small"
                            />
              </div>
</div>,
          ],
          "className": "card flex flex-column w100",
          "style": Object {
            "background": "url() no-repeat center",
            "backgroundSize": "cover",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "aspect-5-4 w100",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div>
                  <span
                                    className="w100 bold mt2 mb1 block"
                  />
                  <p
                                    className="detail mb3"
                  />
</div>,
                <div
                  className="relative"
>
                  <Button
                                    className="small bg-seafoam absolute t0 l0"
                                    color="white-madison-blue-border"
                                    disabled={true}
                                    fullWidth={false}
                                    label="Sold Out"
                                    onClick={[Function]}
                                    shadow={false}
                                    to=""
                                    type="button"
                                    variant="primary-small"
                  />
</div>,
              ],
              "className": "flex flex-column justify-between bg-seafoam h100 relative",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <span
                      className="w100 bold mt2 mb1 block"
/>,
                    <p
                      className="detail mb3"
/>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "",
                      "className": "w100 bold mt2 mb1 block",
                    },
                    "ref": null,
                    "rendered": "",
                    "type": "span",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "",
                      "className": "detail mb3",
                    },
                    "ref": null,
                    "rendered": "",
                    "type": "p",
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Button
                    className="small bg-seafoam absolute t0 l0"
                    color="white-madison-blue-border"
                    disabled={true}
                    fullWidth={false}
                    label="Sold Out"
                    onClick={[Function]}
                    shadow={false}
                    to=""
                    type="button"
                    variant="primary-small"
/>,
                  "className": "relative",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": null,
                    "className": "small bg-seafoam absolute t0 l0",
                    "color": "white-madison-blue-border",
                    "disabled": true,
                    "fullWidth": false,
                    "label": "Sold Out",
                    "onClick": [Function],
                    "shadow": false,
                    "to": "",
                    "type": "button",
                    "variant": "primary-small",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "div",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
